bezier-rs/lib/src/lib.rs

```diff
@@ -574,61 +574,115 @@ impl Bezier {
 		self.apply_transformation(&|point| point + translation)
 	}
 
-	// TODO: Use an `impl Iterator` return type instead of a `Vec`
-	// TODO: Change this to `intersect(&self, other: &Bezier)` to also work on quadratic and cubic segments
-	// TODO: (or keep this and add two more functions that perform the logic, and make the `intersect` function call the correct one)
-	/// Returns a list of points where the provided line segment intersects with the Bezier curve. If the provided segment is colinear with the bezier, zero intersection points will be returned.
-	/// - `line` - A line segment expected to be received in the format of `[start_point, end_point]`.
-	pub fn intersect_line_segment(&self, line: [DVec2; 2]) -> Vec<DVec2> {
-		// Rotate the bezier and the line by the angle that the line makes with the x axis
-		let slope = line[1] - line[0];
-		let angle = slope.angle_between(DVec2::new(1., 0.));
-		let rotation_matrix = DMat2::from_angle(angle);
-		let rotated_bezier = self.apply_transformation(&|point| rotation_matrix.mul_vec2(point));
-		let rotated_line = [rotation_matrix.mul_vec2(line[0]), rotation_matrix.mul_vec2(line[1])];
+	/// Implementation of the algorithm to find curve intersections by iterating on bounding boxes.
+	/// `curve1_t_interval` is used to identify the t values of the original `curve1` that the current iteration is representing.
+	/// Note that the `t` interval for only the first curve is needed since we want to return `t` with repsect to it.
+	fn intersection_curves(curve1: &Bezier, curve1_t_interval: [f64; 2], curve2: &Bezier, intersections: &mut Vec<f64>, error: f64) {
```

Comment:
I think the function name could be clearer. It would be nice if it could somewhat explain the purpose of curve1_t_interval. Maybe something like `subcurve_intersection`. Also, maybe you can rename `curve1_t_interval` to `original_curve1_t_interval` or something so that it's easier to read.

Reply:
Also, maybe we should specify in the function description that the "return value" is actually one of the parameters.

Reply:
I rewrote it to return a vector - any thoughts on the changes?